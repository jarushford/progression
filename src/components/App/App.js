import React, { Component } from 'react'
import '../../main.scss'

export default class App extends Component {
  render() {
    return (
      <div className="app">
      
      </div>
    )
  }
}
