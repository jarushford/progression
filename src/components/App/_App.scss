.app {
  height: 100vh;
  margin: 0;
  background: url('./assets/terminus.jpg');
  background-size: cover;
}