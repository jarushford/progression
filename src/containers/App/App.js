import React, { Component } from 'react'
import { Switch, Route, withRouter } from 'react-router-dom'
import Header from '../../components/Header/Header'
import Home from '../../components/Home/Home'
import Training from '../Training/Training'
import Ascents from '../Ascents/Ascents'
import Projects from '../Projects/Projects'
import ProjectPage from '../../components/ProjectPage/ProjectPage'
import sampleUserData from '../../assets/sampleUserData'
import * as ACTIONS from '../../actions/index'
import { connect }  from 'react-redux';
import '../../main.scss'

class App extends Component {

  componentDidMount() {
    const { getTrainingData, getProjects, getAscents } = this.props
    getTrainingData(sampleUserData.trainingData)
    getProjects(sampleUserData.projects)
    getAscents(sampleUserData.ascents)
  }

  render() {
    const { projects, disciplineBoulder } = this.props

    if (!this.props.projects) {
      return <div>Loading...</div>
    }

    return (
      <div className={`app ${disciplineBoulder && 'boulder'}`}>
        <Header />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/training" component={Training} />
          <Route exact path="/projects" component={Projects} />
          <Route path="/projects/:name" render={({ match }) => {
            const { name } = match.params
            const currentProject = projects.find(project =>
              project.name === name
            )
            return <ProjectPage project={currentProject} />
          }}/>
          <Route path="/ascents" component={Ascents} />
          <Route path="" />
        </Switch>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  projects: state.projects,
  disciplineBoulder: state.disciplineBoulder
})

const mapDispatchToProps = (dispatch) => ({
  getTrainingData: (data) => dispatch(ACTIONS.getTrainingData(data)),
  getProjects: (data) => dispatch(ACTIONS.getProjects(data)),
  getAscents: (data) => dispatch(ACTIONS.getAscents(data))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))
