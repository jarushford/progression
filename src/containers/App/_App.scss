.app {
  height: 100vh;
  margin: 0;
  background: url('./assets/sport.jpg');
  background-size: cover;
  background-position: center;
  transition: 1s ease;
  overflow: scroll;
  position: relative;
  z-index: 0;
}

.boulder {
  background: url('./assets/boulder.jpg');
  background-size: cover;
  background-position-x: 90%;
  transition: 1s ease;
}

.no-user-msg {
  text-align: center;
  width: 100%;
  margin-top: 150px;
  background: $darkGrey;
  color: $accent;
  height: 100px;
  padding-top: 30px;
}