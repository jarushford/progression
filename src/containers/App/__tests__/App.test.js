import React from 'react'
import { shallow } from 'enzyme'
import { mapStateToProps, App } from '../App'

describe('App', () => {
  let mockProjects

  beforeEach(() => {
    mockProjects = [
      {
        name: 'Top Notch',
        grade: 'V13',
        id: 2
      }
    ]
  })
  
  it('should match the snapshot in default case', () => {
    const wrapper = shallow(<App error='Oh no!' projects={mockProjects} disciplineBoulder={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with an error', () => {
    const wrapper = shallow(<App error='Oh no!' projects={mockProjects} disciplineBoulder={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with data', () => {
    const wrapper = shallow(<App error='Oh no!' projects={mockProjects} disciplineBoulder={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with discipline as sport', () => {
    const wrapper = shallow(<App error='Oh no!' projects={mockProjects} disciplineBoulder={false} />)

    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps', () => {
  it('should return an object with the correct props', () => {
    const mockProjects = [
      {
        name: 'Top Notch',
        grade: 'V13'
      }
    ]
    const mockState = {
      projects: mockProjects,
      disciplineBoulder: true,
      error: ''
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(mockState)
  })
})

