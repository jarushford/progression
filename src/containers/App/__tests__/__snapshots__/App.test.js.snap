// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot in default case 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    disciplineBoulder={false}
    error="Oh no!"
    projects={
      Array [
        Object {
          "grade": "V13",
          "id": 2,
          "name": "Top Notch",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Connect(Error) />,
      ],
      "className": "app false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Connect(Error) />,
        ],
        "className": "app false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    disciplineBoulder={false}
    error="Oh no!"
    projects={
      Array [
        Object {
          "grade": "V13",
          "id": 2,
          "name": "Top Notch",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Connect(Error) />,
      ],
      "className": "app false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Connect(Error) />,
        ],
        "className": "app false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    disciplineBoulder={false}
    error="Oh no!"
    projects={
      Array [
        Object {
          "grade": "V13",
          "id": 2,
          "name": "Top Notch",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Connect(Error) />,
      ],
      "className": "app false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Connect(Error) />,
        ],
        "className": "app false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with discipline as sport 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    disciplineBoulder={false}
    error="Oh no!"
    projects={
      Array [
        Object {
          "grade": "V13",
          "id": 2,
          "name": "Top Notch",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Connect(Error) />,
      ],
      "className": "app false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Connect(Error) />,
        ],
        "className": "app false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
